src/mail/MailUtils.js

```diff
@@ -553,10 +553,25 @@ export function moveToInbox(mails: Mail[]): Promise<*> {
 	}
 }
 
-export function exportMails(mails: Mail[]): Promise<void> {
+/**
+ * convert a set of mail to a set of dataFiles in eml format
+ * @param mails array of mails to convert
+ * @returns {Promise<(DataFile|any)[]>}
+ */
+export function mailsToEmlDataFiles(mails: Mail[]): Promise<DataFile[]> {
 	const mapper = mail => load(MailBodyTypeRef, mail.body)
 		.then(body => mailToEmlFile(mail, htmlSanitizer.sanitize(getMailBodyText(body), false).text))
-	const exportPromise = Promise.map(mails, mapper, {concurrency: 5})
+	return Promise.map(mails, mapper, {concurrency: 5})
```

Comment:
We try to get rid of async loading in the client as much as possible because then rate limit mechanisms don't work correctly. We need to lower concurrency here to 1.